File: c++/arrays/printAllPairs.cpp

#include <iostream>
using namespace std;

void printAllPairs(int arr[], int n)
{

    for (int i = 0; i < n; i++)
    {
        int x = arr[i];
        for (int j = i + 1; j < n; j++)
        {
            int y = arr[j];

            cout << x << " ," << y << endl;
        }
        cout << endl;
    }
}

int main()
{

    int arr[] = {10, 20, 30, 40};
    int n = sizeof(arr) / sizeof(int);

    printAllPairs(arr, n);

    return 0;
}

//Time complexity O(N2)   n square

File: javascript/arrays/printAllPairs.js
let arr = [10, 20, 30, 40];

function printAllPairs(arr) {
	let size = arr.length;

	for (let i = 0; i < size; i++) {
		let x = arr[i];

		for (let j = i + 1; j < size; j++) {
			let y = arr[j];

			console.log(`${x}, ${y}`);
		}
	}
}
printAllPairs(arr);
//Time complexity O(N2)   n square
